### PR TITLE
Ux writers edits

### PR DESCRIPTION
**Pull request by madeline-brennan**
Added note to Checking API, User API, and Roles API per https://acrolinx.atlassian.net/browse/DEV-26948
Updated User API per https://acrolinx.atlassian.net/browse/DEV-26961

### CODE REVIEW

**Comment by tkrisztian95** [comment]
Looks good to me. 
You have added the links pointing to the authentication section and fixed some phrasing.

PS: I think this PR should be targeting/merged to the `develop` branch.

**Review by george-haddad — APPROVED** [review]
All good! There were just some minor typos in the data structure section. This also included a comment with URL that was appearing when it should not have. I took the liberty to fix those with a couple of commits.

File: apiary.apib [hunk 1]
```diff
@@ -1309,6 +1311,12 @@ Security:
 * A user can read and update its own data.
 * Only privileged users (admin, for example) can read or update other users' data.
 
+**Note:** Before you get started with the User API, be sure to review the [authentication and authorization](https://acrolinxapi.docs.apiary.io/#introduction/authentication-and-authorization) requirements. 
+
+## Current User [/api/v1/user/self]
```
Comment (tkrisztian95) [inline apiary.apib:1316]: I think this was moved to under the `## User Resource [/api/v1/user]` section not a long time ago, by me. Keeping this here would not show up on the rendered API doc in Apiary.

Note that in line 1661 this is already documented under section `### Get a User (Self) [GET /api/v1/user/self]`

I think we can remove this section from here.

**Review by tkrisztian95 — CHANGES_REQUESTED** [review]

File: apiary.apib (hunk 1 above)
Comment (george-haddad) [reply apiary.apib:1316]: I'll take care of it.

Comment (george-haddad) [reply apiary.apib]: I removed the above highlighted section in favor of the one in 1661 because it includes examples responses ...etc.

Comment (tkrisztian95) [reply apiary.apib]: Thank you!

**Review by tkrisztian95 — APPROVED** [review]

**Pull request merged**